airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -468,3 +494,22 @@ def on_kill(self):
                                              stderr=subprocess.PIPE)
 
                 self.log.info("YARN killed with return code: %s", yarn_kill.wait())
+
+            if self._kubernetes_driver_pod:
+                self.log.info('Killing pod %s on Kubernetes'
+                              .format(self._kubernetes_driver_pod))
+
+                # Currently only instantiate Kubernetes client for killing a spark pod.
+                try:
+                    client = kube_client.get_kube_client()
+                    api_response = client.delete_namespaced_pod(
+                        self._kubernetes_driver_pod,
+                        self._connection['namespace'],
+                        body=client.V1DeleteOptions(),
+                        pretty=True)
+
+                    self.log.info("Spark on K8s killed with response: %s", api_response)
+
+                except Exception as e:
```

Comment:
Can we make this exception less broad?

Reply:
The examples at https://github.com/kubernetes-client/python/tree/master/examples don't have any more specific uses than this, and there is no real documentation.  Can you advise how to except on different conditions here?

Reply:
Code is the goto for documentation, but when I read the example in the docs:
https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md#example-85

I see that they catch an `ApiException`:
```python
try: 
    api_response = api_instance.delete_namespaced_pod(name, namespace, body, pretty=pretty, grace_period_seconds=grace_period_seconds, orphan_dependents=orphan_dependents, propagation_policy=propagation_policy)
    pprint(api_response)
except ApiException as e:
    print("Exception when calling CoreV1Api->delete_namespaced_pod: %s\n" % e)
```

Maybe next steps is to check the actual response of the call, but that's something for a different PR. Cheers